### PR TITLE
Migrated to null safety

### PR DESCRIPTION
**Pull request by lohnn**
Looks like API is kept intact with null safety added.
Feel free to merge if it looks all OK. Else just holler with feedback 👍

### CODE REVIEW

**Comment by elbeicktalat** [comment]
Hey man, good job but something you added as nullable should be not null.
anyway Thank you for the collaboration.

File: lib/res/shared/auth_button.dart [hunk 1]
```diff
@@ -2,18 +2,16 @@
...
 import 'package:auth_buttons/res/shared/shared_button.dart';
 import 'package:flutter/material.dart';
 
 class AuthButton extends StatelessWidget {
-  final Key key;
+  final Key? key;
 
   ///**[onPressed]** is a void function well be called when the button pressed.
-  final VoidCallback onPressed;
+  final VoidCallback? onPressed;
```
Comment (elbeicktalat) [inline lib/res/shared/auth_button.dart:14]: like here the onPressed parameter should be required not nullable.

Comment (lohnn) [reply lib/res/shared/auth_button.dart:14]: Thanks, I missed that one!

**Comment by lohnn** [comment]
Just want to check, do you want me to fix these issues and create a new PR, or are you working on null safety on your own?

**Comment by elbeicktalat** [comment]
> Just want to check, do you want me to fix these issues and create a new PR, or are you working on null safety on your own?

yeah I'm working on it.
Thanks

**Pull request closed without merging**